lib/presentation/crowdactions/crowdaction_view/commitment_item.dart

```diff
@@ -49,9 +49,15 @@ class CommitmentItem extends StatelessWidget {
                       fontWeight: FontWeight.w700,
                     )),
                 const SizedBox(height: 5),
-                SelectableText(
-                  description,
-                  style: CollactionTextStyles.body,
+                SizedBox(
+                  width: 358,
+                  child: Text(
+                    description,
+                    style: CollactionTextStyles.body,
+                    softWrap: true,
+                    maxLines: 2,
+                    overflow: TextOverflow.fade,
+                  ),
```

Comment:
Would it be better if we used SelectableText instead of Text, And also if the width is dynamic, so that the empty space on the right is filled with padding instead of a static width value: "take the whole space, minus 20 width on the right"  

We may need to implement some responsive behavior later, as the commitments table may increase or decrease in size, and the text should adapt accordingly.

Additionally, why is `maxLines` set to 2 and the `overflow` set to fade?

Reply:
@dromerolovo Hi, as far as I know, softwrap part does the trick and SelectableText doesn't have that attribute. I have tried expandable/flexible widgets to force wrap the text but it didn't work with parent widgets and tables. That is why I used a text widget. If you advise me on a better solution I appreciate that. 

I am glad that you mentioned the responsiveness, I noticed that all the values are fixed such as commitment_panel has a fixed size that's why I did 358 pixels. But I can add a media query. of method to add a width value and go over there. I need to minus -121 pixels but I think I got the idea, please correct me if I am wrong. or we can use a package such as sizer : https://pub.dev/packages/sizer which gives us 'sp' values for text font sizes as in Kotlin I believe we can benefit from it.

And lastly, I thought maxlines 2 would be an appropriate length of description and if again the user enters a longer description, I didn't want to break the UI again as overflow. 

Please let me know your comments on the parts above, so I can apply them accordingly.

Reply:
@CaglarKullu wrapping the parent Container (line: 40) widget with Expanded, will make the trick. It's the only change needed to prevent the text to overflow.